src/sidebar/store/modules/annotations.js

```diff
@@ -230,9 +246,10 @@ const reducers = {
         });
         if (newAnn.moderation) {
           const countDelta = action.isFlagged ? 1 : -1;
-          newAnn.moderation = Object.assign({}, annot.moderation, {
-            flagCount: annot.moderation.flagCount + countDelta,
-          });
+          newAnn.moderation = {
+            ...newAnn.moderation,
+            flagCount: newAnn.moderation.flagCount + countDelta,
```

Comment:
In this context `annot.moderation` and `newAnn.moderation` refer to the same object, but due to the `if` check above, TS can see that `newAnn.moderation` is not null.